### ISSUE TITLE
wxWebRequest crashes when wrong URL is provided

### ISSUE THREAD

**Issue by mtangoo** [wxWidgets/wxWidgets#22903]
Hi, 
am using a similar approach to documentation example, so nothing peculiar at all.
wxWidgets:  3.2 stable version
OS: Linux, Ubuntu
Build tools: gcc version 9.4.0 (Ubuntu 9.4.0-1ubuntu1~20.04.1) 

in OnInit method I create a frame, set it as main Window and hide it. Then I create Login dialog and show it to the user. On ctor of the Login dialog I try to download a file from server using wxWebRequest.

```

bool MyApp::OnInit() {
    wxInitAllImageHandlers();

    Mainframe* frame = new Mainframe{nullptr};
    frame->Maximize(true);
    frame->Show(false);
    SetTopWindow(frame);

    LoginDialog dlg(nullptr);
    if(dlg.ShowModal() == wxID_OK) {
        GetTopWindow()->Show(true);
    } else {
        frame->Destroy();
    }

    return true;
}
```

All works fine when I use correct IP/URL but the moment I put wrong URL like `192.168.1.100:3000` I get a crash.
I tried try catch but it will crash anyway. The stack trace is not very helpful for me to figure out where the crash is coming from. I guess it is because it crashes before app starts the main loop.

```
bool MyApp::OnInit() {
    wxInitAllImageHandlers();

    Mainframe* frame = new Mainframe{nullptr};
    frame->Maximize(true);
    frame->Show(false);

    SetTopWindow(frame);

    LoginDialog dlg(nullptr);
    try {
        if(dlg.ShowModal() == wxID_OK) {
            GetTopWindow()->Show(true);
        } else {
            frame->Destroy();
        }
    } catch(std::exception& e){
        std::string msg = e.what();
        wxString error(msg);
        wxLogError(error);
    }
    return true;
}
```

Can someone help me point to correct direction as to what is actually wrong?
```
0  0x0000000000000000  ??    
1  0x00007ffff796a37f  wx_on_channel_event  /home/myname/wxWidgets/src/gtk/evtloop.cpp  136
2  0x00007ffff55a004e  g_main_context_dispatch    
3  0x00007ffff55a0400  ??    
4  0x00007ffff55a06f3  g_main_loop_run    
5  0x00007ffff608712d  gtk_main    
6  0x00007ffff796a01d  wxGUIEventLoop::DoRun  /home/myname/wxWidgets/src/gtk/evtloop.cpp  61
7  0x00007ffff6f1047b  wxEventLoopBase::Run  /home/myname/wxWidgets/src/common/evtloopcmn.cpp  87
8  0x00007ffff79d95da  wxDialog::ShowModal  /home/myname/wxWidgets/src/gtk/dialog.cpp  176
9  0x00005555555903d6  MyApp::OnInit  /home/myname/Hosanna/Desktop/App/src/Core/app.cpp  18
10  0x000055555559124d  wxAppConsoleBase::CallOnInit  /usr/local/include/wx-3.2/wx/app.h  93
11  0x00007ffff6f4c11a  wxEntry  /home/myname/wxWidgets/src/common/init.cpp  487
12  0x00007ffff6f4c236  wxEntry  /home/myname/wxWidgets/src/common/init.cpp  515
13  0x0000555555590731  main  /home/myname/Desktop/App/src/Core/app.cpp  31
```

**Comment by vadz** [comment]
Can you reproduce the error in the webrequest sample? It's really difficult to tell what's going on when you don't show any code actually using `wxWebRequest`.

**Comment by mtangoo** [comment]
>Can you reproduce the error in the webrequest sample? 
I will modify the sample URL and try it out

>It's really difficult to tell what's going on when you don't show any code actually using wxWebRequest.

I get it. Will find time tomorrow to post results. But I didn't do any magic beyond what is in documentation for wxWebRequest

**Comment by mtangoo** [comment]
@vadz 
Thanks for pointing out to the sample. I actually was having too much confidence in the code since I copied and modified from the docs. I took a test with the sample and alas it crashed not. Comparing the code only one thing was done differently, that is, I was not storing `wxWebRequest` object as a member of the class. 

Doing that solved the crash and now all is working as expected. Should we somehow mention this in the documentation or I was just one idiot out there?

I will meanwhile close this since main issue is resolved
